pkg/api/validation/validation.go

```diff
@@ -30,12 +30,21 @@ import (
 	"github.com/golang/glog"
 )
 
+const QualifiedNameErrorMsg string = " have to match [" + util.Dns1123SubdomainFmt + " / ] " + util.Dns1123LabelFmt + " pattern"
```

Comment:
1) You should not need the leading space on these - I looked at it should be added automatically.  If not, that is a bug.

2) Printing regex is better than nothing, but not ideal for some users.  Good enough for now, but we should think more about this eventually.

3) word-smithing:
- change simple ones to "must match regex " + whatever
- change numeric ranges to "must be greater than X and less than Y"
- change all of the size+regex ones to "must be less than %d characters and match regex " + whatever
- host port should be same as port, right?

4) We should use the defined numeric limits, rather than hard-coding, so you may have to switch to 'var' and fmt.Sprintf()

5) Not sure on this: should these just become functions in util/validation.go ?